src/main/java/it/pagopa/pm/gateway/service/async/CcResumeStep1AsyncService.java

```diff
@@ -0,0 +1,145 @@
+package it.pagopa.pm.gateway.service.async;
+
+import it.pagopa.pm.gateway.client.vpos.HttpClient;
+import it.pagopa.pm.gateway.client.vpos.HttpClientResponse;
+import it.pagopa.pm.gateway.dto.creditcard.StepZeroRequest;
+import it.pagopa.pm.gateway.dto.vpos.*;
+import it.pagopa.pm.gateway.entity.PaymentRequestEntity;
+import it.pagopa.pm.gateway.repository.PaymentRequestRepository;
+import it.pagopa.pm.gateway.utils.EcommercePatchUtils;
+import it.pagopa.pm.gateway.utils.VPosRequestUtils;
+import it.pagopa.pm.gateway.utils.VPosResponseUtils;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+import static it.pagopa.pm.gateway.constant.Messages.GENERIC_ERROR_MSG;
+import static it.pagopa.pm.gateway.constant.VposConstant.RESULT_CODE_AUTHORIZED;
+import static it.pagopa.pm.gateway.constant.VposConstant.RESULT_CODE_CHALLENGE;
+import static it.pagopa.pm.gateway.dto.enums.PaymentRequestStatusEnum.*;
+
+@Service
+@Slf4j
+@NoArgsConstructor
+public class CcResumeStep1AsyncService {
+    private static final String CREQ_QUERY_PARAM = "?creq=";
+    public static final String RESULT_CODE_METHOD = "26";
+
+    @Value("${vpos.requestUrl}")
+    private String vposUrl;
+
+    @Autowired
+    private PaymentRequestRepository paymentRequestRepository;
+
+    @Autowired
+    private VPosRequestUtils vPosRequestUtils;
+
+    @Autowired
+    private VPosResponseUtils vPosResponseUtils;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private EcommercePatchUtils vposPatchUtils;
```

Comment:
Prefer constructor injection to field injection

Reply:
changed